### DISCUSSION TITLE
Not launching with Firefox

### DISCUSSION THREAD

**Discussion by morriarty999** [google/model-viewer#3318] [Q&A]
[model.zip](https://github.com/google/model-viewer/files/8372606/model.zip)
Hi,

By making a simple file with a simple 3D model, nothing is displayed with Firefox. There is only the image.

I saw that "webxr in ar-modes" does not work with Firefox and Safari.
But I'm just using the base code to display the model.

This is the basic code :

`<body>
    
    <model-viewer bounds="tight" enable-pan src="crayonjaune.glb" ar ar-modes="webxr scene-viewer quick-look" camera-controls poster="poster.webp" shadow-intensity="0" shadow-softness="0">
      <div class="progress-bar hide" slot="progress-bar">
          <div class="update-bar"></div>
      </div>
    </model-viewer>  
    <script src="script.js"></script>
    
    <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>`

I put the model in the archive.
Thank you for your help.

**Comment by elalish** [comment]
Did you check your JS console for errors? The most likely is that it can't find your GLB. If you can post a link to the live website we can tell for sure.

**Comment by morriarty999** [comment]
I checked the link with the 3D model and it is good. All other browsers (Chrome, Brave, Opera,Edge, Samsung Internet), correctly display the 3D model.

I haven't tried with Safari as I don't have a mac device

**Comment by morriarty999** [comment]
I tried on another computer and it works with Firefox. I also tested with the browser Safari with a IPAD and it works too.

I don't understand why because it doesn't work on my smartphone with Firefox.

**Reply by elalish** [reply]
The JS console is your friend. When you post its contents, the reason will likely become clear.

**Comment by morriarty999** [comment]
This is the error displayed 
![1](https://user-images.githubusercontent.com/97617405/160893422-689a5b6d-cbf4-477d-b04d-c843fb2be14a.jpg)
![2](https://user-images.githubusercontent.com/97617405/160893426-a6ffe3e8-ea51-4616-aa8e-b817957a1086.jpg)

**Reply by elalish** [reply]
Looks like you're using a custom progress bar? I don't think you want to use `event.target`, as that's probably just the MV element. Not sure what `.center-pre-prompt` is, but I'm sure there's another way you can find it.

**Comment by morriarty999** [comment]
Strangely, it works once but if we reload another page with another 3D model there is only the image

Looks like after the first load the script crashes
